src/storm/mesos/logviewer/SocketUrlDetection.java

```diff
@@ -0,0 +1,66 @@
+package storm.mesos.logviewer;
+
+import java.net.InetAddress;
+import java.net.Socket;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.log4j.Logger;
+
+import com.google.common.base.Optional;
+
+public class SocketUrlDetection implements IUrlDetection {
+
+    public static final int DEFAULT_PORT = 8000;
+    
+    public SocketUrlDetection() {
+        setPort(Optional.of(DEFAULT_PORT));
+    }
+    
+    public SocketUrlDetection(int port) {
+        setPort(Optional.of(port));
+    }
+    
+    public SocketUrlDetection(Optional<Integer> port) {
+        setPort(port);
+    }
+    
+    @Override
+    public boolean isReachable() {
+
+        Socket socket       = null;
+        boolean reachable   = false;
+        try {
+            
+            if (LOG.isDebugEnabled())
+                LOG.debug("Checking host " + InetAddress.getLocalHost() + " and port " + getPort() );
+            
+            if (port.isPresent()) {
+                socket      = new Socket(InetAddress.getLocalHost(), getPort().get());
+                reachable   = true;
+            }
+        } catch (Exception e) {
+            // don't care.
+        }
+        finally {
+            IOUtils.closeQuietly(socket);
+        }
+        return reachable;
+    }
+    
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
+    public void setPort(Optional port) {
+        this.port = (Optional<Integer>) port;
+    } 
+    
+    @Override
+    public Optional<Integer> getPort() {
+        return port;
+    }
+    
+    protected Optional<Integer> port;
+ 
+    private static final Logger LOG = Logger.getLogger(SocketUrlDetection.class);
```

Comment:
Yep, the incorrect way of handling private variables which should always be
at the bottom. Software engineers are lemmings by nature, thats how the
whole privates at the top got started..... but if I must.

On Tue, Jun 9, 2015 at 12:44 PM, Timothy Chen <notifications@github.com>
wrote:

> In src/storm/mesos/logviewer/SocketUrlDetection.java
> <https://github.com/mesos/storm/pull/38#discussion_r32034700>:
>
> > +    }
> > +
> > +    @SuppressWarnings({ "rawtypes", "unchecked" })
> > +    @Override
> > +    public void setPort(Optional port) {
> > +        this.port = (Optional<Integer>) port;
> > +    }
> > +
> > +    @Override
> > +    public Optional<Integer> getPort() {
> > +        return port;
> > +    }
> > +
> > +    protected Optional<Integer> port;
> > +
> > +    private static final Logger LOG = Logger.getLogger(SocketUrlDetection.class);
>
> Variables are on the top of the class
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/storm/pull/38/files#r32034700>.
>